experimental/GraphicalModels/src/PhylogeneticAuxiliary.jl

```diff
@@ -0,0 +1,142 @@
+  for l in lvs
+    in_node = inneighbors(graph,l)[1]
+    lvs_cherr = outneighbors(graph, inneighbors(graph,l)[1])
+    if issubset(lvs_cherr, lvs) == 2
+      cherry = append!(cherry, [[Edge(in_node, lvs_cherr[1]), Edge(in_node, lvs_cherr[2])]])
+    end
+  end
+  
+  return unique(cherry)
+end
+
+function root(graph::Graph)
+  n_parents = [length(inneighbors(graph, v)) for v in 1:n_vertices(graph)]
+  return findall(x -> x == 0, n_parents)[1]
+end
+
+
+## COMPUTE EQUIVALENCE CLASSES ##
+# Given the dictionary of parametrization, the functions below compute 
+# the equivalence classes, one time as is, the second one as in the format 
+# on the website: every polynomial is multiplied by 0.25 times the size of the equivalence class. 
+@doc raw"""
+    compute_equivalent_classes(pm::GroupBasedPhylogeneticModel)  
+
+Given the parametrization of a `PhylogeneticModel`, cancel all duplicate entries and return equivalence classes of states which are attached the same probabilities.
+
+# Examples
+```jldoctest
+julia> pm = jukes_cantor_model(graph_from_edges(Directed,[[4,1],[4,2],[4,3]]));
+julia> compute_equivalent_classes(pm)
+
+Dict{Vector{Tuple{Int64, Int64, Int64}}, QQMPolyRingElem} with 5 entries:
+  [(1, 2, 3), (3, 2, 4), (1, 3, 2), (1, 4, 2), (3, 4, 2), (3, 1, 2), (3, 4, … => 1//4*a[1]*b[2]*b[3] + 1//4*a[2]*b[1]*b[3] + 1//4*a[3]*b[1]*b[2] + 1//4*b[1]*b[2]*b[3]
+  [(2, 1, 1), (4, 3, 3), (1, 2, 2), (3, 2, 2), (2, 4, 4), (2, 3, 3), (3, 1, … => 1//4*a[1]*b[2]*b[3] + 1//4*a[2]*a[3]*b[1] + 1//2*b[1]*b[2]*b[3]
+  [(4, 4, 4), (1, 1, 1), (3, 3, 3), (2, 2, 2)]                                => 1//4*a[1]*a[2]*a[3] + 3//4*b[1]*b[2]*b[3]
+  [(3, 1, 3), (3, 2, 3), (4, 3, 4), (1, 3, 1), (1, 4, 1), (4, 1, 4), (4, 2, … => 1//4*a[1]*a[3]*b[2] + 1//4*a[2]*b[1]*b[3] + 1//2*b[1]*b[2]*b[3]
+  [(2, 2, 1), (3, 3, 2), (4, 4, 3), (1, 1, 2), (3, 3, 1), (4, 4, 2), (2, 2, … => 1//4*a[1]*a[2]*b[3] + 1//4*a[3]*b[1]*b[2] + 1//2*b[1]*b[2]*b[3]
+```
+"""
+function compute_equivalent_classes(parametrization::Dict{})
+  #parametrization = probability_map(pm)
+  polys = unique(collect(values(parametrization)))
+  
+  equivalent_keys = []
+  for value in polys
+      eqv_class = [key for key in keys(parametrization) if parametrization[key] == value]
+      #sort!(eqv_class)
+      append!(equivalent_keys, [eqv_class])
+  end
+  equivalenceclass_dictionary = Dict(equivalent_keys[i] => parametrization[equivalent_keys[i][1]] for i in 1:length(equivalent_keys))
+  return equivalenceclass_dictionary
+end
+
+@doc raw"""
+    sum_equivalent_classes(pm::PhylogeneticModel)  
+
+Take the output of the function `compute_equivalent_classes` for `PhylogeneticModel` and multiply by a factor to obtain probabilities as specified on the original small trees database.
+
+# Examples
+```jldoctest
+julia> pm = jukes_cantor_model(graph_from_edges(Directed,[[4,1],[4,2],[4,3]]));
+julia> sum_equivalent_classes(pm)
+
+Dict{Vector{Tuple{Int64, Int64, Int64}}, QQMPolyRingElem} with 5 entries:
```

Comment:
```suggestion
julia> pm = jukes_cantor_model(graph_from_edges(Directed,[[4,1],[4,2],[4,3]]));

julia> sum_equivalent_classes(pm)
Dict{Vector{Tuple{Int64, Int64, Int64}}, QQMPolyRingElem} with 5 entries:
```
wrong whitespace will make the doctest fail (similar at many other places)